xla/backends/gpu/runtime/command_buffer_cmd.cc

```diff
@@ -411,8 +413,54 @@ absl::Status CommandBufferCmdExecutor::RecordUpdate(
 
   // Keep a state associated with commands in the sequence in the state manager.
   CommandBufferCmd::StateManager& state = record_params.state;
+  const BufferAllocations* allocs = execute_params.buffer_allocations;
```

Comment:
`RecordUpdate` can take absl::flat_hash_set<Index> of indices that changes since last command update, and use that as predicate in the for loop (just a set intersection)

Reply:
Fixed